Product Management Service/Code/product-service/src/main/java/com/shopkeeper/productservice/utils/JsonToMapConverter.java

```diff
@@ -0,0 +1,46 @@
+package com.shopkeeper.productservice.utils;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Converter
+public class JsonToMapConverter implements AttributeConverter<Map<String, Object>, String> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JsonToMapConverter.class);
+
+  @Override
+  @SuppressWarnings("unchecked")
```

Comment:
what does this do? And should we be using this?

Reply:
Here we are converting the String object into HashMap object. So, this annotation is used to explicitly tell the compiler that the user is aware of this and it is legal and can be executed at the time of execution.
If we don't use this, the deserialization will still be happening, but it will just suppress/ignore the warning coming from the compiler.